### DISCUSSION TITLE
Image-zoom component, how to keep rounded corners during transition?

### DISCUSSION THREAD

**Discussion by nktnet1** [fuma-nama/fumadocs#830] [Q&A] [answered]
Using the [image-zoom](https://fumadocs.vercel.app/docs/ui/components/image-zoom) component, I'm currently setting the tailwind class `rounded-full` for both the component props and inside zoomInProps:

```tsx
  <ImageZoom
    src={exampleImage}
    className="border-2 rounded-full h-auto w-[220px] my-0"
    zoomInProps={{
      className: "rounded-full"
    }}
    alt="Example Image"
  />
```

This works for the start and end state of the image during the zoom action, but not in the transition (i.e. going from not zoomed to zoom).

Is there any other props that we can use to make the image remain rounded for the full duration? I've consulted the documentation here but couldn't get information on props or how transition is handled:
- https://fumadocs.vercel.app/docs/ui/components/image-zoom

Thank you!

**Comment by Gitstar-OC** [comment]
In the `image-zoom.js` file in `fumadocs-ui/dist/components/image-zoom.js` you can add  border property 

```js
"use client";
import {
  defaultImageSizes
} from "../chunk-UOD2T27N.js";
import "../chunk-MLKGABMK.js";

// src/components/image-zoom.tsx
import Image from "next/image";
import Zoom from "react-medium-image-zoom";
import "../../dist/image-zoom.css";
import { jsx } from "react/jsx-runtime";
function getImageSrc(src) {
  if (typeof src === "string") return src;
  if ("default" in src) return src.default.src;
  return src.src;
}
function ImageZoom({
  zoomInProps,
  children,
  ...props
}) {
  return /* @__PURE__ */ jsx(
    Zoom,
    {
      zoomMargin: 20,
      wrapElement: "span",
      zoomImg: {
        src: getImageSrc(props.src),
        sizes: void 0,
        ...zoomInProps
      },
      children: children ?? /* @__PURE__ */ jsx(Image, { sizes: defaultImageSizes, ...props }), 
     // other properties you want to add 
    }
  );
}
export {
  ImageZoom
};
```

This will change the border while the image is zoomed.

**Reply by nktnet1** [reply]
I don't think there's a property that allows the border radius to be set from within this file.

Looking at the react-medium-image-zoom library that's being used, I do see this discussion here:
- https://github.com/rpearce/react-medium-image-zoom/discussions/395

Not sure if it's even possible in this case then :shrug:. Thanks for answering though!

**Reply by Gitstar-OC** [reply]
Look, I have tried t in the browser and you can add border radius to the image, 
![Screenshot 2024-09-10 144025](https://github.com/user-attachments/assets/09fd8231-b91c-47a9-bb31-ac37ca2d48b0)


You can try adding border radius using just the image tag in the global css file of the app. It stays the same, and it works

```css
img, video {
border-radius: 15px;
} 
```

**Reply by nktnet1** [reply]
Yeah I think that works - do you have a robust way of setting the img tag to have border-radius without using global CSS or editing the file in node_modules or dist?

Or perhaps @fuma-nama could provide a way to pass props to the internally used img tag 👀

**Reply by Gitstar-OC** [reply]
I think that solves the problem, or you will need to dig down into the whole zoom component in the node_modules which is not a great thing as this can do the thing and save your time.

**Reply by nktnet1** [reply]
I want to avoid modifying the global image tag because there are other images that shouldn't be rounded, and modifying the installed code in node_modules is not robust to updates and hard to deploy to different environments for me :(.

**Reply by fuma-nama** [reply]
you can use CSS selectors, the react medium image zoom component has their own `data-` attributes which is supposed to use for styling

**Reply by nktnet1** [reply]
Thanks @fuma-nama - looking at this code here:
- https://github.com/rpearce/react-medium-image-zoom/blob/c8f091f40ea8adc4fe9841d476d16487c55e9a3b/stories/Img.stories.tsx#L201

It looks like we need to set the `classDialog` property, e.g.
```tsx
          <Zoom {...props} classDialog="custom-zoom">
```

but at the moment, there's no way to set the `classDialog` property in the image-zoom component (it's ignored?):
https://github.com/fuma-nama/fumadocs/blob/de85fc0243b367ebd0e48d800d896f8ab2505786/packages/ui/src/components/image-zoom.tsx#L28-L38

---

I can indeed confirm that by adding this to my global.css:
```css
.custom-zoom [data-rmiz-modal-img] {
  border-radius: 9999px;
}
```

and hard-coding the style to fumadoc's image-zoom component:
```tsx
    Zoom,
    {
      zoomMargin: 20,
      wrapElement: "span",
      zoomImg: {
        src: getImageSrc(props.src),
        sizes: void 0,
        ...zoomInProps
      },
      classDialog: 'custom-zoom',     // hard-coding this here
      children: children ?? /* @__PURE__ */ jsx(Image, { sizes: defaultImageSizes, ...props })
    }
```

does indeed work. Would it be possible for you to let the classDialog prop be passed through? Thank you!

**Reply by fuma-nama** [reply]
I would recommend you to clone the component from our repo and modify it (similar to Shadcn UI). 

I'll fix it later.

**Reply by nktnet1** [reply, marked as answer]
Looks like this has been fixed in [fumadocs-ui@13.4.10](https://github.com/fuma-nama/fumadocs/releases/tag/fumadocs-ui%4013.4.10), i.e. the code below 

Example `style.css`:
```css
.rmiz-rounded-full-zoom [data-rmiz-modal-img] {
  border-radius: 9999px;
}
```

Example `component.tsx`:
```tsx
<ImageZoom
  src={exampleImage}
  className="border-2 rounded-full h-auto w-56 my-0"
  alt="Example Image"
  rmiz={{
    classDialog: 'rmiz-rounded-full-zoom'
  }}
/>
```
Now works as expected. Thanks @fuma-nama and @Gitstar-OC!
